### ISSUE TITLE
Failed to load font-face using relative path url

### ISSUE THREAD

**Issue by nicolaslawdune** [Kozea/WeasyPrint#1825]
Hello,

**Context :**
Font-face url doesn't find font storing inside the project.

**Problem detail :**
Structure of my project :
```
/usr/src/app
  -> main.py
  -> fonts
     -> myfontcss.css
     -> myfont.woff2
     -> myfont.woff
```

Content of myfontcss.css :  
```
h1 {
  font-size:50rem;
} 

@font-face {
  font-family: 'MyFont';
  font-style: normal;
  font-weight: 700;
  src: url('myfont.woff2') format('woff2'), url('myfont.woff') format('woff');
  unicode-range: U+0100-024F,U+0259,U+1E00-1EFF,U+2020,U+20A0-20AB,U+20AD-20CF,U+2113,U+2C60-2C7F,U+A720-A7FF;
}
```

Content of main.py : 
```
css_string =  """
@import 'fonts/myfontcss.css';

h1{
    font-family: 'MyFont';
    font-size: 25pt;
    font-weight: 700;
}
"""
css = CSS(string=css_string, font_config=font_config, base_url='/usr/src/app')
```

**Result :**
myfontcss.css file is load because h1 (50rem) font-size is applied, except the font family. 
The problem didn't occur in version 55, and appear since v56.0b1.

If I copy font face inside the css_string variable, it works.

```
css_string =  """ 
h1{
    font-family: 'MyFont';
    font-size: 25pt;
    font-weight: 700;
}

@font-face {
  font-family: 'MyFont';
  font-style: normal;
  font-weight: 700;
  src: url('fonts/myfont.woff2') format('woff2'), url('fonts/myfont.woff') format('woff');
  unicode-range: U+0100-024F,U+0259,U+1E00-1EFF,U+2020,U+20A0-20AB,U+20AD-20CF,U+2113,U+2C60-2C7F,U+A720-A7FF;
}
"""
css = CSS(string=css_string, font_config=font_config, base_url='/usr/src/app')
```

Thank you for your help.

**Comment by liZe** [comment]
Hi!

Your example works for me with version 58.0. To find what’s wrong, you can enable [logging](https://doc.courtbouillon.org/weasyprint/stable/first_steps.html#logging).

One stupid question: do you have `font-family` set on `h1`? (You don’t in your sample.)

**Comment by nicolaslawdune** [comment]
> Hi!
> 
> Your example works for me with version 58.0. To find what’s wrong, you can enable [logging](https://doc.courtbouillon.org/weasyprint/stable/first_steps.html#logging).
> 
> One stupid question: do you have `font-family` set on `h1`? (You don’t in your sample.)

Yes, I forgot to add font-family in my sample. I edited my post.

I enabled log, but there isn't any problems except the parser doesn't support unicode-range property. I remove it but the problem already exists.

A complementary information : 
I use Flask and host my app in a docker container.

**Comment by liZe** [comment]
> I enabled log, but there isn't any problems except the parser doesn't support unicode-range property. I remove it but the problem already exists.

It’s not supported by WeasyPrint, but that’s not where your problem comes from.

I’m really sorry, but I can’t reproduce your problem. Could you send a Docker image that reproduces the problem?

**Comment by nicolaslawdune** [comment]
[weasyprint.sample.zip](https://github.com/Kozea/WeasyPrint/files/10892666/weasyprint.sample.zip)
Here is the docker project.

Change the version of weasyprint 55 to 56.0 or later to reproduce the problem.

My investigations bring me to say that it find the font-face by the relative path even if it doesn't use it, because if I set a fake relative path, logs display that the font cannot be found.

**Comment by liZe** [comment]
The problem is actually caused by `unicode-range` not being supported. As you define the `@font-face` twice in `700.css`, WeasyPrint only uses the fonts defined in the first rule, and thus uses `libre-baskerville-latin-ext-700-normal.woff2` that doesn’t include common Latin characters.

A simple workaround would be to invert the two rules if you only use common Latin characters. Otherwise, you can use only one rule with the woff font (that includes everything).

The real fix is to fix #378.

**Comment by nicolaslawdune** [comment]
Thank you. It fixed using latin font-face only without `unicode-range`.
